keps/sig-network/1111-dynamically-re-configure-lb-node-set/README.md

```diff
@@ -0,0 +1,681 @@
+that is done however, cloud provider implementations will need to point all HCs
+for services of type `LoadBalancer` to `healthCheckNodePort` if the proposed
+feature flag is enabled.
+
+### Risks and Mitigations
+
+#### Risk 
+
+- Feature flag is enabled on a cluster running a service proxy which does not
+  implement the feature. The HCs will fail on all cluster nodes for ETP=cluster
+  services and ingress will be broken.
+
+#### Mitigations
+
+- Don't enable the feature flag if you are not 100 % sure your service proxy
+  supports the new functionality.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+#### Prerequisite testing updates
+
+#### Unit tests
+
+Components in the architecture presented above are already existing (and
+supposedly are already unit tested): kube-proxy's healthz server, kube-proxy's
+"ETP=local" server. We would need to add tests which validate the change in
+behaviour for opening up additional ports for ETP=cluster services and validate
+that the difference in HC logic between ETP services is captured correctly.
+
+#### Integration tests
+
+Kube-proxy is mostly tested via unit tests and e2e, not integration, and this is
+not expected to change.
+
+#### e2e tests
+
+Kubernetes in general needs to extended its load balancing test suite with
+disruption tests, this might be the right effort we need to get that ball
+rolling. Testing would include: validation that HC probe's return a respond
+corresponding to the "state of the Kubernetes cluster world" during different
+disruption scenarios.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature flag for changing the API for ETP=cluster services and
+  include `healthCheckNodePort` for them. This will help testing and graduation
+  to beta and stable.  
+- Changes to data plane components get implemented (Kube-proxy)
+- Changes to control plane components get implemented, but depends on the
+  feature flag being enabled or not.
+- Initial e2e tests completed and enabled
+
+#### Beta
```

Comment:
this can not go to beta without proof that cloud providers are using the new feature

Reply:
I am fine with this statement. But what proof would we need? Is verifying their code enough?

Reply:
yes